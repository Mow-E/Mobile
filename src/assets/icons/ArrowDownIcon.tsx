import * as React from 'react';
import Svg, {SvgProps, Path, Rect} from 'react-native-svg';
import colors from '../../styles/colors';
import {ArrowIconProps, WIDTH_HEIGHT_RATIO} from './ArrowUpIcon';

function SvgComponent(props: SvgProps): JSX.Element {
  return (
    <Svg width={24} height={24} viewBox="0 0 68 98" fill="none" {...props}>
      <Rect
        x={17}
        y={0}
        width={34}
        height={67.2277}
        fill={props.fill}
        stroke={props.stroke}
      />
      <Path
        d="M 34 97 L 67 61 H 1 L 34 97 Z"
        fill={props.fill}
        stroke={props.stroke}
      />
    </Svg>
  );
}

/**
 * An icon that shows an downwards arrow icon.
 */
function ArrowDownIcon({
  size = 24,
  lightColor = false,
}: ArrowIconProps): JSX.Element {
  const strokeColor = lightColor
    ? colors.secondary.light
    : colors.secondary.dark;

  return (
    <SvgComponent
      height={size}
      width={size * WIDTH_HEIGHT_RATIO}
      stroke={strokeColor}
      fill={strokeColor}
    />
  );
}

export default ArrowDownIcon;
