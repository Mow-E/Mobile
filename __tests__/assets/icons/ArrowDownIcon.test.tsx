import 'react-native';
import React from 'react';
import ArrowDownIcon from '../../../src/assets/icons/ArrowDownIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders default correctly', () => {
  const tree = renderer.create(<ArrowDownIcon />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders colored correctly', () => {
  const tree = renderer.create(<ArrowDownIcon colored />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncolored correctly', () => {
  const tree = renderer.create(<ArrowDownIcon colored={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in dark mode correctly', () => {
  const tree = renderer.create(<ArrowDownIcon darkModeInverted />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in different sizes', () => {
  const treeSmall = renderer.create(<ArrowDownIcon size={24} />).toJSON();
  const treeLarge = renderer.create(<ArrowDownIcon size={48} />).toJSON();

  expect(treeSmall).not.toEqual(treeLarge);
});
