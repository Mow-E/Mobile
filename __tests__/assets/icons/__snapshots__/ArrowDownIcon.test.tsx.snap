// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="16.653061224489797"
  fill="#2f3163"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#2f3163"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 16,
      },
    ]
  }
  vbHeight={98}
  vbWidth={68}
  width={16.653061224489797}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      height="67.2277"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      width="34"
      x="17"
      y="0"
    />
    <RNSVGPath
      d="M 34 97 L 67 61 H 1 L 34 97 Z"
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders default correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="16.653061224489797"
  fill="#2f3163"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#2f3163"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 16,
      },
    ]
  }
  vbHeight={98}
  vbWidth={68}
  width={16.653061224489797}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      height="67.2277"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      width="34"
      x="17"
      y="0"
    />
    <RNSVGPath
      d="M 34 97 L 67 61 H 1 L 34 97 Z"
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders in dark mode correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="16.653061224489797"
  fill="#2f3163"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#2f3163"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 16,
      },
    ]
  }
  vbHeight={98}
  vbWidth={68}
  width={16.653061224489797}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      height="67.2277"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      width="34"
      x="17"
      y="0"
    />
    <RNSVGPath
      d="M 34 97 L 67 61 H 1 L 34 97 Z"
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders uncolored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="16.653061224489797"
  fill="#2f3163"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#2f3163"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 16,
      },
    ]
  }
  vbHeight={98}
  vbWidth={68}
  width={16.653061224489797}
>
  <RNSVGGroup
    fill={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4281282915,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      height="67.2277"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      width="34"
      x="17"
      y="0"
    />
    <RNSVGPath
      d="M 34 97 L 67 61 H 1 L 34 97 Z"
      fill={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4281282915,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
